main/src/ca/uwaterloo/flix/language/phase/Kinder.scala

```diff
@@ -1193,6 +1195,16 @@ object Kinder {
       }
   }
 
+  private def visitJvmMethod(method: ResolvedAst.JvmMethod, kenv: KindEnv, senv: Map[Symbol.UnkindedTypeVarSym, Symbol.UnkindedTypeVarSym], taenv: Map[Symbol.TypeAliasSym, KindedAst.TypeAlias], henv: Option[(Type.KindedVar, Type.KindedVar)], root: ResolvedAst.Root)(implicit flix: Flix) = method match {
+    case ResolvedAst.JvmMethod(ident, fparams, exp, _, purAndEff, loc) =>
+      val fparamsVal = traverse(fparams)(visitFormalParam(_, kenv, senv, taenv, root))
+      val expVal = visitExp(exp, kenv, senv, taenv, henv, root)
+      val purAndEffVal = visitPurityAndEffect(method.purAndEff, kenv, senv, taenv, root)
+      val tpeVal = Type.freshVar(Kind.Star, loc.asSynthetic)
```

Comment:
I'm not convinced that the above is correct (I'm ignoring type type that's in `ResolvedAst.JvmMethod`?).

If this is right, can someone explain to me what's going on. And if it's wrong, what should it be instead?

Reply:
Indeed, you should be visiting the type of the JvmMethod node. Something like `val tpeVal = visitType(tpe, Kind.Star, <all the envs>)`